### PR TITLE
Add support for querying for latest version in catalog

### PR DESCRIPTION
**Pull request by aclevername**
### Description

You can now provide "latest" as the argument to `version`

### Checklist
- [ ] Added tests that cover your change
- [ ] Added/modified documentation as required (such as the `README.md` (diagrams, usage, roadmap etc), and anything in `examples/`)
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] Added a `kind` label to the PR (e.g. `kind/feature`)

### CODE REVIEW

File: pkg/catalog/catalog.go [hunk 1]
```diff
@@ -62,15 +64,47 @@ func (c *Catalog) Get(sourceName, profileName string) *profilesv1.ProfileDescrip
...
+}
+
+func getLatestVersion(profiles []profilesv1.ProfileDescription, profileName string) *profilesv1.ProfileDescription {
+	var profilesWithValidVersion []profileDescriptionWithVersion
+
+	for _, p := range profiles {
+		v, err := version.NewVersion(p.Version)
+		if err != nil {
+			continue
+		}
```
Comment (aclevername) [inline pkg/catalog/catalog.go:97]: I think we should validate a catalog entry has a valid semver version somewhere else, so here it just ignores the profile if its not valid

Comment (Callisto13) [reply pkg/catalog/catalog.go:97]: where is the somewhere else?

Comment (aclevername) [reply pkg/catalog/catalog.go:97]: ideally a validation webhook, to give the error back to the user at `kubectl create` time

Comment (Callisto13) [reply pkg/catalog/catalog.go:97]: cool 👍

**Review by Callisto13 — APPROVED** [review]

**Pull request merged**